### ISSUE TITLE
batch processing process time increase not expected 

### ISSUE THREAD

**Issue by bigPacha** [ZhengPeng7/BiRefNet#106]
Hello thank you for birefnet its really well trained
I wanted to test inference of  birefnet with torchserve (to process many images as batches, but the process time increase according to the number of images passed in the batch )
I was expecting process time for many images passed in the batch  should be little similar than processing one image  

here is an example of code

```
imgs = [img ] 
batch_images = torch.stack([transform_image(im) for im in imgs])
batch_images = batch_images.to(device)

time_start = time.time()
with torch.no_grad():
    preds = birefnet(batch_images)[-1].sigmoid().cpu()
time_end = time.time()

print('prediciton time 1 img ==>', time_end -time_start)


imgs = [img , img] 
batch_images = torch.stack([transform_image(im) for im in imgs])
batch_images = batch_images.to(device)

time_start = time.time()
with torch.no_grad():
    preds = birefnet(batch_images)[-1].sigmoid().cpu()
time_end = time.time()

print('prediciton time 2 img ==>', time_end -time_start)
```

and here is the output

I tested on L4 nvidia (24 GB of gpu memory) 
so im little confused

here is the output of processing time (in Seconds)

prediciton time before warmup ==> 0.9029507637023926
prediciton time 1 img ==> 0.32914161682128906
prediciton time 2 img ==> 0.7225363254547119
prediciton time 3 img ==> 1.0903816223144531
prediciton time 4 img ==> 1.4586782455444336
prediciton time 5 img ==> 1.9846432209014893

**Comment by ZhengPeng7** [comment]
Hi, @bigPacha. I'm so sorry that I don't know why I didn't receive the notification of your issue. I was just checking the issue list to see if some problem was unsolved and saw your issue.

Do you mean the inference time is increasing linearly, though it's in a batch process?
Actually, I had the same thoughts as yours. However, after some practical trials, I find that it's only a theoretical calculation. In many cases where the computation cost is very high (like here), even if the GPU memory is not occupied fully, it might have already reached the bottleneck. The `GPU-Util` is more reliable than `Memory-Usage` for the measurement, as shown in `nvidia-smi`.
